File: python/programmers/level 2/짝지어 제거하기.py
def solution(s):
    stack = []
    for letter in s:
        if len(stack) != 0:
            if letter == stack[-1]:
                stack.pop()
            else:
                stack.append(letter)
        else:
            stack.append(letter)
    if len(stack) > 0:
        return 0
    return 1

# def solution(s):
#     copy_s = s
#     for letter in copy_s:
#         s = s.replace(letter+letter, '')
#     if len(s) == 0:
#         return 1
#     return 0

# def solution(s):
#     answer = -1
#     index = 0
#     while index < len(s) - 1:
#         if s[index] == s[index+1]:
#             s = s[:index] + s[index+2:]
#             index -= 2
#         index += 1
#         if index < 0:
#             index = 0
#     if len(s) == 0:
#         return 1
#     return 0

File: java/src/main/java/ps_traning/programmers/level_2/짝지어_제거하기.java
package ps_traning.programmers.level_2;

import java.util.Stack;

public class 짝지어_제거하기 {
    public int solution(String s) {
        int answer = 0;
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (stack.size() == 0 || stack.peek() != c) {
                stack.push(c);
            } else {
                stack.pop();
            }
        }
        if (stack.size() == 0) {
            answer = 1;
        }
        return answer;
    }
}
